app/src/main/res/values/styles.xml

```diff
@@ -411,11 +411,8 @@
   </style>
   <!-- Start Over Lesson Button -->
   <style name="SecondaryButton" parent="TextAppearance.AppCompat.Widget.Button">
-    <item name="android:layout_width">144dp</item>
+    <item name="android:layout_width">0dp</item>
```

Comment:
If someone were to change the parent layout of these buttons to something other than ConstraintLayout, would they still render correctly?

Reply:
Unfortunately, I don't think it will work if we change to another type of layout. Since those buttons are pretty much defined by the constraints we set to dynamically fit different screen sizes. I test a bit about this number and I found both 0dp and 144dp have the same result. The constraint has higher precedence than those fixed numbers so those numbers basically been ignored.

Reply:
I do think having these as 0dp or match_constraint would effectively resolve this issue since the buttons will always scale to fit the screensize. However defaulting it at such a high level in styles could lead to the button not displaying at all with other uses of the button (both now and in the future), hence the suggestion to move the attributes to the view level.

Reply:
Thank you so much, the explanation is really helpful. I will move attributes to the corresponding place in the fragment layout.